### PR TITLE
Extensions: WPSC: Add Debug State

### PR DESCRIPTION
**Pull request by ockham**
Split off #16026 to keep things somewhat smaller. Requires an up-to-date `master` branch of WPSC.

To test:
Not wired yet, so just verify that unit tests pass.

### CODE REVIEW

File: client/extensions/wp-super-cache/state/debug/reducer.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * @param  {Object} state Current debug logs
+ * @param  {Object} action Action object
+ * @return {Object} Updated debug logs
+ */
+export const items = createReducer( {}, {
+	[ WP_SUPER_CACHE_REQUEST_DEBUG_LOGS_SUCCESS ]: ( state, { siteId, data } ) => ( { ...state, [ siteId ]: data } ),
+	[ WP_SUPER_CACHE_DELETE_DEBUG_LOG_SUCCESS ]: ( state, { siteId, filename } ) => ( {
+		...state,
+		[ siteId ]: omit( state[ siteId ], filename )
```
Comment (donnapep) [inline client/extensions/wp-super-cache/state/debug/reducer.js]: Ah, so `filename` is there temporarily while deleting, and then removed. Nice.

File: client/extensions/wp-super-cache/state/debug/reducer.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	} )
+} );
+
+export default combineReducers( {
```
Comment (donnapep) [inline client/extensions/wp-super-cache/state/debug/reducer.js]: You don't want to add little 'ole `requesting`? I think he feels left out. ☹️ We could add a test for that as well if we wanted to ensure the expected keys are there.

File: client/extensions/wp-super-cache/state/debug/test/selectors.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+	const primarySiteId = 123456;
+	const secondarySiteId = 456789;
+	const filename = '89fe3b92191d36ee7fb3956cd52c704c.php';
+
+	describe( 'getDebugLogs()', () => {
+		const primaryDebugLogs = {
+			'89fe3b92191d36ee7fb3956cd52c704c.php': '145cc79483f018538a3edc78117622ba'
+		};
+
+		it( 'should return null if no state exists', () => {
```
Comment (donnapep) [inline client/extensions/wp-super-cache/state/debug/test/selectors.js]: `null` or 'empty object'?

File: client/extensions/wp-super-cache/state/debug/test/selectors.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+						}
+					}
+				}
+			};
+			const isDeleting = isDeletingDebugLog( state, secondarySiteId, filename );
+
+			expect( isDeleting ).to.be.false;
+		} );
+
+		it( 'should return false if the cache is not being deleted', () => {
```
Comment (donnapep) [inline client/extensions/wp-super-cache/state/debug/test/selectors.js]: 'cache' or 'debug log'?

File: client/extensions/wp-super-cache/state/debug/test/selectors.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+						}
+					}
+				}
+			};
+			const isDeleting = isDeletingDebugLog( state, primarySiteId, filename );
+
+			expect( isDeleting ).to.be.false;
+		} );
+
+		it( 'should return true if the cache is being deleted', () => {
```
Comment (donnapep) [inline client/extensions/wp-super-cache/state/debug/test/selectors.js]: Oh, the dangers of copy / paste. 😆

File: client/extensions/wp-super-cache/state/debug/reducer.js (hunk 2 above)
Comment (ockham) [inline client/extensions/wp-super-cache/state/debug/reducer.js]: Good catch, thanks!

Comment (ockham) [inline client/extensions/wp-super-cache/state/debug/reducer.js]: af29bc5121b5d0d5e5ea4b98591baa8e4a508221 and 6c9d15f345cb44c3237a8af87b2ce6cc15897079

**Pull request merged**